[FreeBSD] date, expand the command, simplify blurb

grahamperrin opened the pull request:
releng/13.0 died on 31st (not 16th) August 2022.

<https://www.freebsd.org/security/unsupported/>

The blurb is simplified, to include only what's immediately relevant to the (supported version) rows in the table that precedes the blurb. Readers can follow the additional links, to learn more, if required.

freebsd-version alone (run without options) does not tell whether the running kernel is patched, following installation of a patch. So: 

freebsd-version -kru

– and refer to the manual page.

grahamperrin committed:
[FreeBSD] fix dates, update links and simplify description

- Update versionCommand : freebsd-version alone (run without options) does not tell whether the running kernel is patched.
- Simplify description to include only what's immediately relevant to the (supported version) rows in the table that precedes the blurb. Readers can follow the links to learn more, if required.

grahamperrin commented on products/freebsd.md:
```diff
@@ -6,27 +6,31 @@ releasePolicyLink: https://www.freebsd.org/security/#sup
 -   releaseCycle: "releng/13.1"
     eol: false
     # 13.2-RELEASE + 3 months
-    link: https://www.freebsd.org/releases/13.1R/announce/
```
In the **Release** column at <https://endoflife.date/freebsd>, the hint when pointing at a link is: 

> Release Notes / Changelog for …

`/announce/` URLs are for announcements – **not** release notes. 

The shorter form of URL allows the reader to more easily see the **direct link** to release notes, alongside links to other pages that are relevant to the release.

usta replied:
https://www.freebsd.org/releases/13.1R/ -> https://www.freebsd.org/releases/13.1R/relnotes/
same for others

grahamperrin replied:
> [freebsd.org/releases/13.1R](https://www.freebsd.org/releases/13.1R/) -> [freebsd.org/releases/13.1R/relnotes](https://www.freebsd.org/releases/13.1R/relnotes/) 

Usually, a direct link to release notes would be a first choice. 

Unusually, FreeBSD release notes **lack the date of release** (and I do expect some interest in dates in the `endoflife.date` site context). 

The shorter form of URL allows the reader to see, _at a glance_: 

* the link to release notes
* the link to the schedule, which includes the release date

– and so on.

marcwrobel replied:
I agree with @grahamperrin for this particular case.

grahamperrin replied:
Thanks. 

On the FreeBSD side: https://reviews.freebsd.org/D37528 aims for future improvements.

usta replied:
Then lets keep it like @grahamperrin offers, but we still need a proper changelogtemplate to overcome having lots of link tags @marcwrobel

marcwrobel replied:
`changelogTemplate` added.

Link for stable release must be set manually : I do not see a way to build a `changelogTemplate` compatible with those cycles.

marcwrobel commented on products/freebsd.md:
```diff
@@ -134,14 +146,11 @@ releases:
+* whenever there is a new `RELEASE` from a `stable` branch, [Security Officer](https://www.freebsd.org/administration/#t-secteam) support for the prior `RELEASE` from the same branch will end three months later.
 
-Each FreeBSD Release has an associated Security Branch. The Security Branch tags have names like `releng/10.1`. The corresponding builds have names like `FreeBSD 10.1-RELEASE-p4`.
+[freebsd-version(1)](https://www.freebsd.org/cgi/man.cgi?query=freebsd-version&sektion=1&manpath=FreeBSD)
```
I think we should keep the definition of `stable` and `releng`. I do not know FreeBSD and I had to look at https://www.freebsd.org/security/#sup to know what `releng` means.

marcwrobel replied:
@grahamperrin, @usta, I added a commit to :

- Revert the product description to what can be found on https://www.freebsd.org/. Usually the product description is a copy-paste of what can be found on the product landing page.
- Simplify blurb by combining what I found on https://www.freebsd.org/security/#model (with an explanation of what is `stable/x` and what is `releng/x.y`) and a small paragraph explaining that the EOL date are not exact. All the details mentioning the security officers have been removed: if EndOfLife users want details they can go to the _More information_ link : https://www.freebsd.org/security/.

What do you think of this new description ?

grahamperrin replied:
> … to know what `releng` means.

A good question. 

----

[FreeBSD Release Engineering](https://docs.freebsd.org/en/articles/freebsd-releng/) – an _article_ in the documentation portal – begins by describing itself as outdated. 

Neither the _Documentation_ menu, nor the footer, includes a link to FAQ. This should be fixed. 

[Release Engineering Information](https://www.freebsd.org/releng/) – a _FreeBSD Project_ page – is better, but not ideal. There's mention of `stable` and `releng`, without explaining that `stable` is (for many use cases) **not** best for stability; and so on. 

Again, no direct link to FAQ; it's nested in the _Documentation_ menu. I'd like to change this, to raise awareness of existence of FAQ, but not just yet (see below, the rewrite). 

The [home page](https://www.freebsd.org/) does include a direct link to FAQ, however the design of the page causes links such as these to be overlooked. 

Existing FreeBSD bugs include [265276 – Books: FAQ: Single HTML: 404](https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=265276).

The FAQ page mentions _release engineering_ but not `releng`. This should be fixed. 

Generally: 

* an **entire rewrite of the FAQ page** is planned :-)

----

Improvements to the main **Get FreeBSD** page (a.k.a. the huge yellow button at the home page) are ongoing. 

Whilst we're without a suitable FAQ page, two improvements last month ([1](https://github.com/freebsd/freebsd-doc/commit/5c8cc20333c4e87dd5d3a562ad87f5c18254d5fd?diff=split#diff-0b0b77802642b4009de6cc60f986b1f3d3dc56593e50ca3b1d41dae75da5402cR36-R38), [2](https://github.com/freebsd/freebsd-doc/commit/5c8cc20333c4e87dd5d3a562ad87f5c18254d5fd?diff=split#diff-0b0b77802642b4009de6cc60f986b1f3d3dc56593e50ca3b1d41dae75da5402cR144-R160)) may be of interest. Resulting texts: 

* <https://www.freebsd.org/where/#download>

> RELEASE versions of FreeBSD are the end result of _release engineering_.

* <https://www.freebsd.org/where/#helptest>

– includes an outline of the three branches. I could have added lowercase `main`, `stable` and `releng`, however (a) the page is already too long for my liking; and (b) a better place will be the rewritten FAQ.

marcwrobel replied:
Thanks for the explanation @grahamperrin. Feel free to revisit the description to add better link(s) once they are available.

grahamperrin commented on products/freebsd.md:
```diff
@@ -146,11 +146,11 @@ releases:
-* whenever there is a new `RELEASE` from a `stable` branch, [Security Officer](https://www.freebsd.org/administration/#t-secteam) support for the prior `RELEASE` from the same branch will end three months later.
-
-[freebsd-version(1)](https://www.freebsd.org/cgi/man.cgi?query=freebsd-version&sektion=1&manpath=FreeBSD)
+The Expected EoL (end-of-life) column indicates the earliest date on which support for that branch or release will end.
```
There's a table with columns at <https://endoflife.date/freebsd>. 

> The Expected EoL (end-of-life) column

Readers will naturally seek this column in the table that's present. It's not there.

marcwrobel replied:
You are right, it should have been called _Security Support_. That's fixed.

grahamperrin commented on products/freebsd.md:
```diff
@@ -146,11 +146,11 @@ releases:
-
-[freebsd-version(1)](https://www.freebsd.org/cgi/man.cgi?query=freebsd-version&sektion=1&manpath=FreeBSD)
+The Expected EoL (end-of-life) column indicates the earliest date on which support for that branch or release will end.
+Please note that these dates may be pushed back if circumstances warrant it.
```
Superfluous. 

The possibility of change is inherent to a date being _expected_ (neither _definite_ nor _final_).

marcwrobel replied:
With _Security Support_ is becomes valid again.

grahamperrin commented on products/freebsd.md:
```diff
@@ -146,11 +146,11 @@ releases:
-* whenever there is a new `RELEASE` from a `stable` branch, [Security Officer](https://www.freebsd.org/administration/#t-secteam) support for the prior `RELEASE` from the same branch will end three months later.
-
-[freebsd-version(1)](https://www.freebsd.org/cgi/man.cgi?query=freebsd-version&sektion=1&manpath=FreeBSD)
+The Expected EoL (end-of-life) column indicates the earliest date on which support for that branch or release will end.
```
Was removal of the link to the manual page intentional? 

The manual page is, debatably, amongst the most important things – if a reader is to understand _why_ the options are given in the suggested command.

marcwrobel replied:
Yes, it was, because it was a bit strange to see that information was not next to the command itself (which is not possible without modifying the template used to generate this page).

The purpose of the command (on endoflife.date) is to give information about the current FreeBSD version, not about anything else. On second thought I think we should remove these options if that's OK with you.

grahamperrin commented on products/freebsd.md:
```diff
@@ -146,11 +146,11 @@ releases:
+> [FreeBSD](https://www.freebsd.org/) is an operating system used to power modern servers, desktops, and embedded platforms.
 
-[The support model](https://www.freebsd.org/security/#sup), summarised:
+Under the [current support model](https://www.freebsd.org/security/#sup), each major version’s stable branch
```
No need to mention that the model is _current_. It was established more than seven years ago, and is not likely to change in the foreseeable future.

marcwrobel replied:
You are right, fixed.

grahamperrin reviewed (commented):
Changes, please. 

As background: various FreeBSD Project pages are undergoing improvement (I'm a committer to the documentation tree, which includes [the website](https://github.com/freebsd/freebsd-doc/tree/main/website/)). 

Better focus on directing readers to descriptions at the Project website – with the possibility/likelihood of changes there – than to adhere too closely when copying and paraphrasing texts that may become outdated. 

Thanks

grahamperrin committed:
freebsd.md

Improve the description. 

No need to mention modernity. 

Avoid confusion about support for point releases. (For example: 13.1 was followed by 12.4, this does not mean that support for 13.1 will end three months after 12.4 was released.)

Conciseness. Stating 'the earliest date' (not definitely stating 'the date') implies the possibility of change.
